### DISCUSSION TITLE
How to bring back the focus to the previous focused control ?

### DISCUSSION THREAD

**Discussion by Whiletru3** [AvaloniaUI/Avalonia#17611] [Q&A]
Hello,

Is there a way to save the previous focused element and bring it back at lost focus of the current one?

Like in Ms Word, you type a text in the document, then you change the focus to the font family combo and you type the name of the font you want. When you validate the combo, the focus is back to the text at the right place.

**Comment by msneijders** [comment]
I think redirecting focus on a `LostFocus` is not common; if something is losing focus, something else is getting focus? In your Ms Word example: after setting font type, the user might want to set focus to "font size" -- this will trigger a 'lost focus' on font type. Redirecting focus to text at that moment is probably not what the user expects.

I think Word is setting focus [explicitly](https://docs.avaloniaui.net/docs/concepts/input/focus#explicit-focusing) to the text after font type has changed (and also after font size has changed etc).

If you want to fiddle around with lost/gotfocus yourself, try this in your `MainWindow`:

```C#
public class MyMainWindow
{
   static MyMainWindow()
   {
      // see https://docs.avaloniaui.net/docs/concepts/input/routed-events#class-handlers:
      Avalonia.Input.InputElement.GotFocusEvent.AddClassHandler<MyMainWindow>((x, e) => x.OnGotFocus(e));
      Avalonia.Input.InputElement.LostFocusEvent.AddClassHandler<MyMainWindow>((x, e) => x.OnLostFocus(e));
   }

   InputElement _previousFocused;

   void OnGotFocus(Avalonia.Input.GotFocusEventArgs args)
   {
     // remember args.Source (should be of type InputElement)
    _previousFocused = args.Source is InputElement x ? x : _previousFocused;
   }

   void OnLostFocus(Avalonia.Interactivity.RoutedEventArgs args)
   {
      _previousFocused.Focus(); // but I doubt this will do what you want
   }
}
```

But as I said, setting focus explicitly on a change by the user is probably more stable.

**Reply by thevortexcloud** [reply]
There is also focus manager that does some things you may want.

https://docs.avaloniaui.net/docs/concepts/services/focus-manager
